src/components/List/List.tsx

```diff
@@ -0,0 +1,58 @@
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import { twMerge } from 'tailwind-merge';
+import type { FlowbiteStateColors } from '../..';
+import { mergeDeep } from '../../helpers/merge-deep';
+import { getTheme } from '../../theme-store';
+import type { DeepPartial } from '../../types';
+import { ListItem } from './ListItem';
+
+export interface FlowbiteListTheme {
+  root: FlowbiteListRootTheme;
+}
+
+export interface FlowbiteListRootTheme {
+  base: string;
+  ordered: {
+    on: string;
+    off: string;
+  };
+  unstyled: string;
+  nested: string;
+}
+
+export interface ListColors extends FlowbiteStateColors {
+  [key: string]: string;
+  default: string;
+}
+
+export interface ListProps extends PropsWithChildren<ComponentProps<'ul'> & ComponentProps<'ol'>> {
+  theme?: DeepPartial<FlowbiteListTheme>;
+  ordered?: boolean;
+  unstyled?: boolean;
+  nested?: boolean;
+}
+
+const ListComponent: FC<ListProps> = ({ children, className, theme: customTheme = {}, ...props }) => {
+  const theme = mergeDeep(getTheme().list, customTheme);
+  const Component = props.ordered ? 'ol' : 'ul';
+
+  return (
+    <Component
+      className={twMerge(
+        theme.root.ordered[props.ordered ? 'on' : 'off'],
```

Comment:
I think it would be a good idea for the future to start using https://cva.style/docs for the variants, instead of handling it differently for each component.